tflint/runner.go

```diff
@@ -205,6 +207,32 @@ func NewModuleRunners(parent *Runner) ([]*Runner, error) {
 	return runners, nil
 }
 
+// Check runs inspection by the passed rule.
+// Although it is a wrapper for rule.Check, it properly determines whether the rule is enabled.
+func (r *Runner) Check(rule Rule) error {
+	enabled := rule.Enabled()
+	if r := r.Config.Rules[rule.Name()]; r != nil {
+		if r.Enabled {
+			log.Printf("[DEBUG] `%s` is enabled", rule.Name())
+		} else {
+			log.Printf("[DEBUG] `%s` is disabled", rule.Name())
+		}
+		enabled = r.Enabled
+	}
+
+	if enabled {
+		return rule.Check(r)
+	}
+	return nil
+}
```

Comment:
From one side I don't like that rules depend on Config, which is passed through Runner. From the other side, in the end it is easy to track who Enables and Disables rules.